### PR TITLE
Add BetterTransformer to HF Bert Encoder

### PR DESCRIPTION
**Pull request by erichan1**
Steps to use the benchmark:

Install PyTorch 1.12.1 to get access to the fast path of nn.TransformerEncoder.

Install this version of HF transformers by cloning this branch and running this cmd from the transformers dir
`pip install -e` .

Navigate to the benchmarking folder, uncomment the desired configs in run_sweep.sh, and run `./run_sweep.sh`
The times, comparisons, and numerical tests will log to log.csv.

### CODE REVIEW

File: src/transformers/models/bert/modeling_bert.py [hunk 1]
```diff
@@ -526,13 +638,120 @@ def feed_forward_chunk(self, attention_output):
...
 
 class BertEncoder(nn.Module):
     def __init__(self, config):
         super().__init__()
         self.config = config
         self.layer = nn.ModuleList([BertLayer(config) for _ in range(config.num_hidden_layers)])
         self.gradient_checkpointing = False
+        self.use_fastpath = False
+    
+    def to_torch(self):
```
Comment (erichan1) [inline src/transformers/models/bert/modeling_bert.py]: to_torch is purely for testing against slow path of nn.TransformerEncoder

**Comment by erichan1** [comment]
Closing because this won't be landed. Only used as a reference for HF integration.

**Comment by tanmaylaud** [comment]
Hi, may I know what happened to this integration process?
any status update?

**Comment by erichan1** [comment]
> Hi, may I know what happened to this integration process? any status update?

It's been delayed. We are planning this for later. cc @HamidShojanazeri

**Comment by erichan1** [comment]
@tanmaylaud We did it. See https://github.com/huggingface/transformers/issues/18374.

**Pull request closed without merging**